x-pack/legacy/plugins/ml/server/routes/job_service.js

```diff
@@ -4,11 +4,37 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import Boom from 'boom';
 import { callWithRequestFactory } from '../client/call_with_request_factory';
+import { callWithInternalUserFactory } from '../client/call_with_internal_user_factory';
 import { wrapError } from '../client/errors';
 import { jobServiceProvider } from '../models/job_service';
+import { categorizationExamplesProvider } from '../models/job_service/new_job';
+import { isSecurityDisabled } from '../lib/security_utils';
+
+export function jobServiceRoutes({
+  commonRouteConfig,
+  elasticsearchPlugin,
+  route,
+  xpackMainPlugin,
+}) {
+  async function hasPermissionToCreateJobs(callWithRequest) {
+    if (isSecurityDisabled(xpackMainPlugin) === true) {
+      return true;
+    }
+
+    const resp = await callWithRequest('ml.privilegeCheck', {
```

Comment:
As an alternative to doing the privilege check using the `_has_privileges` API, you can augment the [reserved privilege](https://github.com/elastic/kibana/blob/9fd75dbb919bcde924bef712c04efd356bb13d9a/x-pack/legacy/plugins/ml/server/new_platform/plugin.ts#L171) which is assigned to both the `machine_learning_admin` and `machine_learning_user` roles and take advantage of the "API Authorization" similar to this example: https://www.elastic.co/guide/en/kibana/current/development-plugin-feature-registration.html#_example_2_dev_tools

Reply:
Have spoken to @kobelb and have agreed to leave this as is for now as upcoming spaces work will see the replacement of all privilege checks in ML.